src/plugins/controls/server/options_list/options_list_cluster_settings_route.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { get } from 'lodash';
+
+import { getKbnServerError, reportServerError } from '@kbn/kibana-utils-plugin/server';
+import { CoreSetup } from '@kbn/core/server';
+
+export const setupOptionsListClusterSettingsRoute = ({ http }: CoreSetup) => {
+  const router = http.createRouter();
+  router.get(
+    {
+      path: '/api/kibana/controls/optionsList/getClusterSettings',
+      validate: false,
+    },
+    async (context, _, response) => {
+      try {
+        const esClient = (await context.core).elasticsearch.client.asCurrentUser;
+        const allowExpensiveQueries = get(
+          await esClient.cluster.getSettings({
+            include_defaults: true,
+            filter_path: '**.allow_expensive_queries',
+          }),
+          'transient.search.allow_expensive_queries'
```

Comment:
nit: Here you're only searching for the transient setting of `search.allow_expensive_queries`, there is actually a [hierarchy of settings](elastic.co/guide/en/elasticsearch/reference/current/cluster-update-settings.html). I was just looking, and I think we can make this more robust by [getting](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-get-settings.html) the settings in a different format. Maybe if you `include_defaults` you won't need to use the ternary. We might have to do some research on this.

Reply:
Fixed in 380de4d21eaef98c14dfb0f259191825d9a6c615 after advice from the Core team :+1: